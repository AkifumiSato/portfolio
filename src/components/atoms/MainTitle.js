import React from 'react'
import styled  from 'styled-components'

const MyTitle = styled.h1`
  display: flex;
  flex-direction: column;
  font-size: 40px;
  font-weight: bold;
  height: 250px;
  justify-content: center;
  text-align: center;
  @media screen and (max-width: 768px) {
    font-size: 30px;
    height: 150px;
    padding: 0;
  }
`

const Category = styled.span`
  display: block;
  font-size: 20px;
  @media screen and (max-width: 768px) {
    font-size: 16px;
  }
`

const MainText = styled.span`
  display: block;
  font-size: 40px;
  margin-top: 10px;
  @media screen and (max-width: 768px) {
    font-size: 24px;
  }
`

const MainTitle = ({ title, category = '' }) => category ? (
  <MyTitle>
    <Category>{ category }</Category>
    <MainText>{ title }</MainText>
  </MyTitle>
) : (
  <MyTitle>{ title }</MyTitle>
)

export default MainTitle
