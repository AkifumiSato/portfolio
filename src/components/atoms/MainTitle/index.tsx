import * as React from 'react'
import styled from 'styled-components'
import color from '../../../styles/color'
import { ralewayMixin } from '../../../styles/font'

const MyTitle = styled.h1`
  display: block;
  ${ ralewayMixin };
  font-size: 40px;
  font-weight: bold;
  padding-bottom: 10px;
  border-bottom: 1px solid ${ color.gray.base };

  @media screen and (max-width: 768px) {
    font-size: 24px;
  }
`

const MainTitle: React.FC = ({ children }) => (
  <MyTitle>
    { children }
  </MyTitle>
)

export default MainTitle
