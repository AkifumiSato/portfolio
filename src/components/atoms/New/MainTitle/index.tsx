import * as React from 'react'
import styled from 'styled-components'
import { colors } from '../../../../styles/color'

const MyTitle = styled.h1`
  font-size: ${({ fontSize }: SizeProperty) => fontSize};
  color: ${colors.teal['900']};
  font-weight: normal;
  line-height: 1.2;
  padding-bottom: 50px;
  border-bottom: 1px solid ${colors.darkBlue['300']};

  @media screen and (max-width: 768px) {
    font-size: 24px;
  }
`

type SizeProperty = {
  fontSize: string
}

type SizeDictionary = {
  large: SizeProperty
  normal: SizeProperty
}

const sizeProperty: SizeDictionary = {
  large: {
    fontSize: '50px',
  },
  normal: {
    fontSize: '40px',
  },
}

type SizeProps = keyof SizeDictionary

const MainTitle: React.FC<{ size?: SizeProps }> = ({
  size = 'large',
  children,
}) => {
  const { fontSize } = sizeProperty[size]
  return <MyTitle fontSize={fontSize}>{children}</MyTitle>
}

export default MainTitle
