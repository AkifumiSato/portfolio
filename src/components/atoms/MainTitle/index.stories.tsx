import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info'
import MainTitle from './index'

storiesOf(MainTitle.name, module)
  .addDecorator((story, context) => withInfo({
    text: `記事のメインタイトルなどに使用します。`
  })(story)(context))
  .add('Single Title',() => <MainTitle title="Single Title" />)
  .add('Multi Title',() => <MainTitle title="Single Title" category="BLOG" />)

