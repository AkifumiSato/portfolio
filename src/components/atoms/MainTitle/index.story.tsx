import { storiesOf } from '@storybook/react'
import * as React from 'react'
import MainTitle from '.'

storiesOf(`atom|${ MainTitle.name }`, module)
  .add(
    'Single Title',
    () => <MainTitle title="Single Title" />,
    {
      info: `
        上位の階層のタイトルに使用します。
      `
    })
  .add(
    'Multi Title', () =>
      <MainTitle title="Multi Title" category="BLOG" />,
    {
      info: `
        下位の階層のタイトルに使用します。
      `
    })
