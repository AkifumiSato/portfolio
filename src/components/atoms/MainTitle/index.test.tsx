import * as React from "react"
import * as renderer from "react-test-renderer"

import MainTitle from "./index"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<MainTitle title="Multi Title" category="BLOG" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})