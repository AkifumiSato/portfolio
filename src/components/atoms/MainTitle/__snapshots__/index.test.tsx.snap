// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<h1
  className="sc-bdVaJa gUyIKe"
>
  <span
    className="sc-bwzfXH dKffac"
  >
    BLOG
  </span>
  <span
    className="sc-htpNat buhAHh"
  >
    Multi Title
  </span>
</h1>
`;
